src/camera.py

```diff
@@ -186,78 +186,102 @@ def get_tilt_angle(self):
+                                       + self._tilt_offset,
+                                       self.MIN_PAN_TILT_DEGREE,
+                                       self.MAX_TILT_DEGREE))
+
+
 def main():
     """
     Demonstrates the camera's panning and tilting functionality.
 
-    Creates a camera object and pans from 0 to 180, then from 180 to 0. The
-    camera then tilts from 0 to 150, then 150 to 0.
+    Creates a camera object using absolute degrees and pans from 0 to 180, then
+    from 180 to 0. The camera then tilts from 0 to 150, then 150 to 0.
+    Then creates a camera object using relavtive degrees and pans from 0 to 180,
+    then from 180 to 0. The camera then tilts from 0 to 150, then 150 to 0.
     """
     camera_delay = 0.05
+    camera_delay_long = 0.10
     picar.setup()
 
     # Creates a Camera object with absolute degrees (default).
     camera = Camera()
-
     print "Testing absolute camera degrees."
+
+    # Moving the camera to the right for panning.
+    camera.pan(0)
+    sleep(camera_delay_long)
+
     print "Panning left."
     for i in range(0, 181, 2):
         camera.pan(i)
-        time.sleep(camera_delay)
+        sleep(camera_delay)
 
     print "Panning right."
     for i in range(180, -1, -2):
         camera.pan(i)
-        time.sleep(camera_delay)
+        sleep(camera_delay)
 
     # Moving the camera to the center for tilting.
-    camera.pan(90)
-    time.sleep(camera_delay)
+    camera.reset_camera()
+    sleep(camera_delay_long)
 
     print "Tilting up."
     for i in range(0, 151, 2):
         camera.tilt(i)
-        time.sleep(camera_delay)
+        sleep(camera_delay)
 
     print "Tilting down."
     for i in range(150, -1, -2):
         camera.tilt(i)
-        time.sleep(camera_delay)
-
-    # Resetting camera for relative panning and tilting.
-    camera.pan(0)
-    time.sleep(camera_delay)
+        sleep(camera_delay)
 
     # Creates a Camera object with relative degrees.
     camera = Camera(camera_type=RELATIVE)
     print "Testing relative camera degrees."
 
-    # Moving the camera to the far right before panning left.
+    # Moving the camera to the right for panning.
     camera.pan(-90)
+    sleep(camera_delay_long)
 
     print "Panning left."
     for i in range(90):
         camera.pan(2)
-        time.sleep(camera_delay)
+        sleep(camera_delay)
 
     print "Panning right."
     for i in range(90):
         camera.pan(-2)
-        time.sleep(camera_delay)
+        sleep(camera_delay)
 
-    # Return camera to the center.
-    camera.pan(90)
-    time.sleep(camera_delay)
+    # Moving the camera to the center for tilting.
+    camera.reset_camera()
+    sleep(camera_delay_long)
 
     print "Tilting up."
     for i in range(75):
         camera.tilt(2)
-        time.sleep(camera_delay)
+        sleep(camera_delay)
 
     print "Tilting down."
     for i in range(75):
         camera.tilt(-2)
-        time.sleep(camera_delay)
+        sleep(camera_delay)
```

Comment:
Depending on what you need to do, maybe we can find a different solution instead of the individual sleeps

Reply:
Without sleeps the physical hardware cannot keep up with python, if both delay variables were set to 0 the main functions would look incredibly different and would be unable to display the abilities of the camera class.